python/lsst/meas/extensions/scarlet/deconvolveExposureTask.py

```diff
@@ -0,0 +1,301 @@
+        inputs["band"] = inputRefs.coadd.dataId["band"]
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(
+        self,
+        coadd: afwImage.Exposure,
+        band: str,
+        peaks: PeakCatalog | None = None,
+        **kwargs
+    ) -> pipeBase.Struct:
+        # Load the scarlet lite Observation
+        observation = self._buildObservation(coadd, band)
+        # Initialize the model
+        scarletModel = self._initializeModel(observation, peaks)
+
+        # Iteratively deconvolve the image
+        scarletModel.fit(
+            max_iter=self.config.maxIter,
+            e_rel=self.config.eRel,
+            min_iter=self.config.minIter,
+        )
+
+        # Store the model in an Exposure
+        model = scarletModel.get_model().data[0]
+        image = afwImage.Image(
+            array=model,
+            xy0=coadd.getBBox().getMin(),
+            deep=False,
+            dtype=coadd.image.array.dtype,
+        )
+        maskedImage = afwImage.MaskedImage(
+            image=image,
+            mask=coadd.mask,
+            variance=coadd.variance,
+            dtype=coadd.image.array.dtype,
+        )
+        exposure = afwImage.Exposure(
+            maskedImage=maskedImage,
+            exposureInfo=coadd.getInfo(),
+            dtype=coadd.image.array.dtype,
+        )
+        return pipeBase.Struct(deconvolved=exposure)
+
+    def _removeHighFrequencySignal(
+        self, coadd: afwImage.Exposure
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Remove high frequency signal from the image and PSF.
+
+        This is done by performing a wavelet decomposition of the image
+        and PSF, setting the high frequency wavelets to zero, and
+        reconstructing the image and PSF from the remaining wavelets.
+
+        Parameters
+        ----------
+        coadd : `afwImage.Exposure`
+            Coadd image to deconvolve
+
+        Returns
+        -------
+        image : `np.ndarray`
+            Low frequency image
+        psf : `np.ndarray`
+            Low frequency PSF
+        """
+        psf = coadd.getPsf().computeKernelImage(coadd.getBBox().getCenter()).array
+        # Build the wavelet coefficients
+        wavelets = scl.detect.get_wavelets(
+            coadd.image.array[None, :, :],
+            coadd.variance.array[None, :, :],
+            scales=self.config.waveletScales,
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets.
+        # This has the effect of preventing high frequency noise
+        # from interfering with the detection of peak positions.
+        wavelets[0] = 0
+        # Reconstruct the image from the remaining wavelet coefficients
+        image = scl.wavelet.starlet_reconstruction(
+            wavelets[:, 0],
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets from the PSF.
```

Comment:
So that's the insight that I had while writing this algorithm. The first order starlets are the difference between convolving the image with the starlet kernel, so removing the high frequency signal just leaves the starlet convolved with the image. I did a quick writeup yesterday and today to describe the algorithm and how it works and Ive attached it to the ticket.

Reply:
Alright.. I don't have an intuition for wavelets or starlets so I only half-understood, but other people on the project do so fair enough.